src/lsm/set_associative_cache.zig

```diff
@@ -128,14 +129,21 @@ pub fn SetAssociativeCache(
         clocks: PackedUnsignedIntegerArray(Clock),
 
         pub fn init(allocator: mem.Allocator, value_count_max: u64) !Self {
-            assert(math.isPowerOfTwo(value_count_max));
+            const sets = @divExact(value_count_max, layout.ways);
+
+            switch (value_count_max_requirement) {
+                .power_of_2 => {
+                    assert(math.isPowerOfTwo(value_count_max));
+                    assert(math.isPowerOfTwo(sets));
+                },
```

Comment:
Could we remove this constraint for all caches? I don't understand what the power-of-2 buys us.